tutorials/plugins/gdnative/gdnative-cpp-example.rst

```diff
@@ -31,20 +34,37 @@ There are a few prerequisites you'll need:
 - a Godot 3.x executable,
 - a C++ compiler,
 - SCons as a build tool,
-- a copy of the `godot_headers repository <https://github.com/GodotNativeTools/godot_headers>`_,
 - a copy of the `godot-cpp repository <https://github.com/GodotNativeTools/godot-cpp>`_.
 
 See also :ref:`Compiling <toc-devel-compiling>` as the build tools are identical
 to the ones you need to compile Godot from source.
 
 You can download these repositories from GitHub or let Git
 do the work for you.
-Note that these repositories now have different branches for different versions of Godot. GDNative modules written for an earlier version of Godot will work in newer versions (with the exception of one breaking change in ARVR interfaces between 3.0 and 3.1) but not vise versa so make sure you download the correct branch.
+Note that these repositories now have different branches for different 
+versions of Godot. GDNative modules written for an earlier version of 
+Godot will work in newer versions (with the exception of one breaking 
+change in ARVR interfaces between 3.0 and 3.1) but not vise versa so 
+make sure you download the correct branch.
+Also note that the version of Godot you use to generate the ``api.json``
+with becomes your minimum version.
 
 If you are versioning your project using Git,
 it is a good idea to add them as Git submodules:
 
-.. code-block:: none
+.. tabs::
+ .. code-tab:: none Godot
+
+    mkdir gdnative_cpp_example
+    cd gdnative_cpp_example
+    git init
+    git submodule add https://github.com/GodotNativeTools/godot-cpp
+    cd godot-cpp
```

Comment:
I don't think you need to `cd` into the folder to run `git submodule` commands.
You can simplify the four last lines with just `git submodule update --init`.

Reply:
Same for the 3.0 code block of course.

Reply:
Hmm not sure about that one, you won't believe how often I've had to tell people they need to run those commands in that folder. That said, you can only hold someones hand so much

Reply:
I'm pretty sure it's not necessary :)
```
[akien@cauldron tmp]$ mkdir gdnative_cpp_example
[akien@cauldron tmp]$ cd gdnative_cpp_example/
[akien@cauldron gdnative_cpp_example]$ git init
Dépôt Git vide initialisé dans /home/akien/tmp/gdnative_cpp_example/.git/
[akien@cauldron gdnative_cpp_example (master)]$ git submodule add https://github.com/GodotNativeTools/godot-cpp
Clonage dans '/home/akien/tmp/gdnative_cpp_example/godot-cpp'...
remote: Enumerating objects: 107, done.
remote: Counting objects: 100% (107/107), done.
remote: Compressing objects: 100% (84/84), done.
remote: Total 2881 (delta 26), reused 57 (delta 22), pack-reused 2774
Réception d'objets: 100% (2881/2881), 2.66 MiB | 2.89 MiB/s, fait.
Résolution des deltas: 100% (1988/1988), fait.
[akien@cauldron gdnative_cpp_example (master)]$ git submodule update --init
[akien@cauldron gdnative_cpp_example (master)]$ git submodule 
 bcc39bbf4bf9922c1a5eaea91885eda05cd1202c godot-cpp (v1.0.0-alpha2-94-gbcc39bb)
[akien@cauldron gdnative_cpp_example (master)]$ ls godot-cpp/
binding_generator.py  godot_headers/  include/  LICENSE  Makefile  misc/  README.md  SConstruct  src/
```

Reply:
Oh nevermind, I see the problem. There's also a submodule in the submodule, that's why.

Reply:
So yeah, either keep it as is, or use `git submodule update --init --recursive` (which I tested and I can confirm it works).